.github/workflows/dotnet-test.yml

```diff
@@ -34,21 +44,28 @@ jobs:
 
   analyze:
     name: Analyze
-    runs-on: ubuntu-latest
+    runs-on: windows-latest
```

Comment:
Any reason why we change the os?

Reply:
Another great question! I touched on it briefly in the description of the PR but not in very much detail.

The gist is that this os declaration was actually misleading before since it actually ran on three different OSes (ubuntu, macOS, and windows) due to the "matrix" declaration a few lines down. And we wouldn't have changed anything if it wasn't for the fact that we noticed that CodeQL messed with our Azure PowerShell tasks specifically on the macOS runner. Digging into it further, we came to the conclusion that CodeQL doesn't actually need to run on all OSes since it's a code inspection tool as opposed to one that inspects build outputs. So I ended up just picking windows to run CodeQL on.

Reply:
Testing it on Linux, it's actually much faster. So I'm going to convert it to ubuntu.